src/EffectiveCSharp.Analyzers/PreferDeclarationInitializersToAssignmentStatementsAnalyzer.cs

```diff
@@ -0,0 +1,637 @@
+            InitializerExpressionSyntax initializerExpressionSyntax => IsExpressionValidDeclarationInitialization(initializerExpressionSyntax.Expressions, context, isConstructorParameterLess),
+            BinaryExpressionSyntax binaryExpressionSyntax => IsExpressionValidDeclarationInitialization(binaryExpressionSyntax.Left, context, isConstructorParameterLess) && IsExpressionValidDeclarationInitialization(binaryExpressionSyntax.Right, context, isConstructorParameterLess),
+            _ => false,
+        };
+    }
+
+    private static bool IsExpressionValidDeclarationInitialization(InvocationExpressionSyntax invocation, SyntaxNodeAnalysisContext context, bool isConstructorParameterless)
+    {
+        return isConstructorParameterless && IsExpressionValidDeclarationInitialization(invocation.ArgumentList, context, isConstructorParameterless) &&
+               IsStaticMethodInvocation(invocation, context);
+    }
+
+    // Overload for SyntaxNode
+    private static bool IsExpressionValidDeclarationInitialization(ArgumentListSyntax? argumentList, SyntaxNodeAnalysisContext context, bool isConstructorParameterless)
+    {
+        if (argumentList is null)
+        {
+            return false;
+        }
+
+        foreach (SyntaxNode node in argumentList.ChildNodes())
+        {
+            if (node is not ArgumentSyntax argument)
+            {
+                continue;
+            }
+
+            if (argument.Expression is IdentifierNameSyntax
+                || !IsExpressionValidDeclarationInitialization(argument.Expression, context, isConstructorParameterless))
+            {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    // Overload for SeparatedSyntaxList<ExpressionSyntax>
+    private static bool IsExpressionValidDeclarationInitialization(SeparatedSyntaxList<ExpressionSyntax> expressions, SyntaxNodeAnalysisContext context, bool isConstructorParameterLess)
+    {
+        foreach (ExpressionSyntax expression in expressions)
+        {
+            if (!IsExpressionValidDeclarationInitialization(expression, context, isConstructorParameterLess))
+            {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /// <summary>
+    /// Checks whether the invocation expression is a static method invocation.
+    /// </summary>
+    /// <param name="invocationExpression">The target <see cref="InvocationExpressionSyntax"/>.</param>
+    /// <param name="context">The <see cref="SyntaxNodeAnalysisContext"/>.</param>
+    /// <returns>A bool on whether the invocation is for a static method.</returns>
+    private static bool IsStaticMethodInvocation(InvocationExpressionSyntax invocationExpression, SyntaxNodeAnalysisContext context)
```

Comment:
This is the only way of knowing if the method is static, since the only other way is via the declaration. Only static methods can be used to initialize fields in declarations. The analyzer does not work to help you understand whether the method you are calling SHOULD be static or not.